source/core/GenericFunctions.cpp

```diff
@@ -1614,11 +1614,27 @@ AQBlock1* ArrayToArrayCopyHelper(TypeRef elementType, AQBlock1* pDest, IntIndex*
 static AQBlock1* ArrayToArrayCopyHelperRev(TypeRef elementType, AQBlock1* pDest, IntIndex* destSlabLengths,
     AQBlock1 *pSource, IntIndex* sourceDimLengths, IntIndex* sourceSlabLengths, Int32 destRank, Int32 sourceRank) {
     if (sourceRank == 1) {
-        if (elementType->CopyData(pSource, pDest, sourceDimLengths[0]))
-            return nullptr;
-        Int32 copiedLength = sourceDimLengths[0] * elementType->TopAQSize();
-        if (copiedLength < destSlabLengths[1]) {
-            memset(pDest + copiedLength, 0, destSlabLengths[1] - copiedLength);
+        Int32 topAQSize = elementType->TopAQSize();
+        Int32 copyElems = sourceDimLengths[0];
+        Int32 copyLength = copyElems * elementType->TopAQSize();
+        pDest += copyLength;
+        pSource += copyLength;
+        while (copyElems > 0) {
```

Comment:
Why not use a for loop that runs copyElems times?

Reply:
Because this is exactly equivalent, and the initialization section of the for loop would be empty (or have two statements with a comma operator in it).